### ISSUE TITLE
radarall misses objects of equal distance

### ISSUE THREAD

**Issue by CyborgMaster** [colobot/colobot#816]
`radarall` uses a map of distance to object to store them in while searching, which used to work fine when you only needed to return one (because if two were exactly the same distance away, then ether one would be correct), but now that we have `radarall` in addition to `radar` this needs to be fixed to make sure it gets all the objects.  

This normally would be an extremely rare occurrence (trying to get two floating point distances to be exactly the same (especially with rounding errors) is almost impossible), except for at the very start of the level.  When the level first starts, I want to count the resources we brought with us on the spaceship.  At the beginning your robot is placed in one corner of the ship, and often 4 resources are placed in a square at the opposite corner.  This causes two of those resources to be exactly equidistant to your robot, so `radarall` returns 3, instead of 4.  

The layout ends up looking something like this:
```
o = PowerCell
x = Center of Spacehship
B = robot

o     o

o     o


            x



                     B   
```

The code responsible is in object_manager.cpp inside of `RadarAll`, line 285:
``` cpp
    std::map<float, CObject*> best;
```

I'll give upgrading it a shot and open a PR if I am successful.

**Comment by CyborgMaster** [comment]
So... that was significantly simpler that I assumed.  I only had to change a `map` to a `multimap` to handle duplicate keys.  PR here: #817

**Comment by melex750** [comment]
Maybe it solves this bug too?

Converter just converted titanium...

```int types[] = {Converter, Titanium};```
```object[] obj = radarall(types);```   // will return the same object twice. (Two Titaniums)

**Comment by CyborgMaster** [comment]
I don't think so.  If it is returning two objects, that must mean that they have different distances to the robot.  If they had the same distance, I could see it returning one object (either the Titanium or the Converter), when it should have returned two, but I don't see how this bug would have duplicated the one object and returned it twice.

**Comment by melex750** [comment]
I think you're right, the bug I described happened a long time ago, and I can't reproduce it now,
it was probably something I did.

**Comment by krzys-h** [comment]
Fixed in 88227a3647e88508d817af236b81aeb4d5e522d0
